### PR TITLE
TableView: Add Selection support

### PR DESCRIPTION
**Pull request by bshepherdson**

### CODE REVIEW

File: src/foam/u2/TableView.js [hunk 1]
```diff
@@ -46,6 +49,14 @@ foam.CLASS({
   methods: [
     function addObj(obj) {
       var e = this.start('tr')
+          .on('click', function() {
+            this.tableView.selection = obj;
```
Comment (adamvy-google) [inline src/foam/u2/TableView.js]: Is there a way we could attach this listener only once at the table element?  Otherwise we have to make a DOM access for every row which could be slow when scrolling.

Comment (bshepherdson) [reply src/foam/u2/TableView.js]: In principle, yes. In practice, scrolling is still instant, so I'm not sure it's worth the complexity of determining the target row (and corresponding object) from the click event.

Comment (adamvy-google) [reply src/foam/u2/TableView.js]: Is scrolling instant on mobile/arm chromebooks?

Comment (bshepherdson) [reply src/foam/u2/TableView.js]: It's fast on my phone (although since we're binding directly to the wheel events, it doesn't work for touch except on the scrollbar).

I don't have an underpowered ARM Chromebook handy to try it on. It's acceptable on this underpowered Intel one, though it could still definitely be faster.

Comment (kgrgreer) [reply src/foam/u2/TableView.js]: Each extra listener is 11k. So if you're scrolling 100 rows at 60fps that's ~ 10k_100rows_60frames = 60000 or 60M / sec

Comment (bshepherdson) [reply src/foam/u2/TableView.js]: You can't scroll anywhere near that fast in practice, but your point is taken.

If we're concerned about generating garbage here, I'll write the grungy logic to figure out which row was clicked and which object it corresponds to, and use that instead.

Comment (kgrgreer) [reply src/foam/u2/TableView.js]: I wouldn't hold up the CL for this, but add a TODO to try using a single listener for performance/memory.

Comment (bshepherdson) [reply src/foam/u2/TableView.js]: Already done, PTAL at the new code.

**Comment by adamvy-google** [comment]
LGTM

File: src/com/google/foam/demos/olympics/Controller.js [hunk 2]
```diff
@@ -126,16 +126,15 @@ foam.CLASS({
         scroller.value += negative ? -rows : rows;
       });
       this.container_ = parent.start().cssClass(this.myCls('table-container'));
-      this.container_.start(this.TableView, {
+      this.table = this.container_.start(this.TableView, {
```
Comment (adamvy-google) [inline src/com/google/foam/demos/olympics/Controller.js]: You can use the .call(function() { self.table = this; }) style as well here.

File: src/foam/u2/search/FilterController.js [hunk 3]
```diff
@@ -248,11 +251,14 @@ foam.CLASS({
...
-      parent.start(this.TableView, {
+      // NB: This function should render the table into the parent Element, and
+      // should also set this.table to be the TableView.
+      this.table = parent.start(this.TableView, {
         data$: this.filteredDAO$,
         editColumnsEnabled: true,
         title$: this.title$
-      }).end();
+      });
+      this.table.end();
```
Comment (kgrgreer) [inline src/foam/u2/search/FilterController.js]: Why is this line needed?

File: src/com/google/foam/demos/olympics/Controller.js (hunk 2 above)
Comment (bshepherdson) [reply src/com/google/foam/demos/olympics/Controller.js]: I've restructured this whole block to be fully fluent, using the `.call()` style.

File: src/foam/u2/search/FilterController.js (hunk 3 above)
Comment (bshepherdson) [reply src/foam/u2/search/FilterController.js]: I've restructured this to use the `.call()` style instead.

**Review by bshepherdson — COMMENTED** [review]
PTAL

**Comment by adamvy-google** [comment]
LGTM
